query/math.go

```diff
@@ -42,13 +42,70 @@ func (v *MathValuer) Value(key string) (interface{}, bool) {
 
 func (v *MathValuer) Call(name string, args []influxql.Expr) (interface{}, bool) {
 	if len(args) == 1 {
+		arg0, ok := v.getFloatValue(args[0])
+		if !ok {
+			return nil, false
+		}
+		
+		var fn func(float64) float64
 		switch name {
+		case "abs":
+			fn = math.Abs
```

Comment:
I think we need to be more careful with this. Most of these functions operate the same as the trigonometry functions. For example, `log()` and its derivatives should always return a float. I don't think it makes sense for them not to. But some of these methods it doesn't necessarily make sense. For example, if I have `abs(value::integer)` I don't think it makes sense for it to return a float. I also don't think `pow(value::integer, 2)` makes sense to return a float either. If I were a user, I would expect using two integers with `pow()` to return an integer. On the other hand, I would expect `sqrt(x)` to always return a float since I don't think there's any scenario where it makes sense for it not to return a float.

So not all of these can reuse that code. On first glance, I see `abs()` and `pow()` as the only two I would apply that integer rule to. The others should always return floats in my opinion.